libcudacxx/include/cuda/std/__cccl/builtin.h

```diff
@@ -86,16 +86,60 @@
 #  define _CCCL_BUILTIN_ARRAY_EXTENT(...) __array_extent(__VA_ARGS__)
 #endif // _CCCL_HAS_BUILTIN(__array_extent)
 
-#if _CCCL_HAS_BUILTIN(__builtin_assume_aligned) || _CCCL_COMPILER(MSVC, >=, 19, 23) || _CCCL_COMPILER(GCC)
-#  define _CCCL_BUILTIN_ASSUME_ALIGNED(...) __builtin_assume_aligned(__VA_ARGS__)
-#endif // _CCCL_HAS_BUILTIN(__builtin_assume_aligned)
+#if _CCCL_CHECK_BUILTIN(builtin_acos) || _CCCL_COMPILER(GCC)
+#  define _CCCL_BUILTIN_ACOSF(...) __builtin_acosf(__VA_ARGS__)
+#  define _CCCL_BUILTIN_ACOS(...)  __builtin_acos(__VA_ARGS__)
+#  define _CCCL_BUILTIN_ACOSL(...) __builtin_acosl(__VA_ARGS__)
+#endif // _CCCL_CHECK_BUILTIN(builtin_acos)
+
+#if _CCCL_CUDA_COMPILER(CLANG)
+#  undef _CCCL_BUILTIN_ACOSF
+#  undef _CCCL_BUILTIN_ACOS
+#  undef _CCCL_BUILTIN_ACOSL
+#endif // _CCCL_CUDA_COMPILER(CLANG)
```

Comment:
Why this way, instead of adding `&& !_CCCL_CUDA_COMPILER(CLANG)` to the condition around where these are defined?

Reply:
In this case its relatively simple, but we have a ton of situations where the enablement of the builtin has multiple conditions and mixing a chain of || with a chain of `&& !` just got ugly fast.

So we settled on splitting it into enablement and "compiler issues". That should also make it easier to spot what is a compiler issue and what is actually part of the condition

Reply:
If this is a compiler *issue*, we really should have a comment somewhere in there recording what it is.